cryptarchia/cryptarchia.py

```diff
@@ -1,21 +1,34 @@
-
+from typing import TypeAlias, List, Optional
+from hashlib import sha256
 
 # Please note this is still a work in progress
 from dataclasses import dataclass
 
 Id: TypeAlias = bytes
 
+
 @dataclass
 class Epoch:
     # identifier of the epoch, counting incrementally from 0
     epoch: int
 
+
+@dataclass
+class TimeConfig:
+    # How many slots in a epoch, all epochs will have the same number of slots
+    slots_per_epoch: int
+    # How long a slot lasts in seconds
+    slot_duration: int
+    # Start of the first epoch, in unix timestamp second precision
+    chain_start_time: int
+
+
 # An absolute unique indentifier of a slot, counting incrementally from 0
 @dataclass
 class Slot:
     absolute_slot: int
 
-    def from_unix_timestamp_s(config: TimeConfig, timestamp_s: int) -> Date:
+    def from_unix_timestamp_s(config: TimeConfig, timestamp_s: int) -> "Slot":
```

Comment:
what are the quotes for?

Reply:
You can't use refer to the same class name in a type signature, this is a limitation of python's interpreter in that it evaluates the class body before it the class symbol is ready for use.